File: app/src/main/java/com/example/overtone/ChordGroupDetailsFrag.java
package com.example.overtone;

import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.example.overtone.data.ChordGroup;
import com.example.overtone.data.MusicItem;

import java.util.ArrayList;
import java.util.Arrays;

public class ChordGroupDetailsFrag extends Fragment {

    // TODO: Rename parameter arguments, choose names that match
    // the fragment initialization parameters, e.g. ARG_ITEM_NUMBER
    private static final String ARG_PARAM1 = "param1";
    private static final String ARG_PARAM2 = "param2";

    private ArrayList<ChordGroup> chordGroups;
    private ChordGroup sentChordGroup;

    // TODO: Rename and change types of parameters
    private String mParam1;
    private String mParam2;

    public ChordGroupDetailsFrag() {
        // Required empty public constructor
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (getArguments() != null) {
            mParam1 = getArguments().getString(ARG_PARAM1);
            mParam2 = getArguments().getString(ARG_PARAM2);
        }
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_group_details, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        retrievePassedFragmentData();
//        TextView tView = view.findViewById(R.id.centerTextWords);
//        tView.setText(sentChordGroup.getDescription() + ", " + sentChordGroup.getGroupMakeUp());


    }

    public void retrievePassedFragmentData(){
        if(getArguments()!=null){
            ChordGroupDetailsFragArgs args = ChordGroupDetailsFragArgs.fromBundle(getArguments());
            //receiving chord group data object that was clicked on
              this.sentChordGroup = args.getChordGroupClickedOn();
        }
    }

}

File: app/src/main/java/com/example/overtone/ChordGroupDetailsFrag.kt
package com.example.overtone

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.navigation.NavController
import androidx.navigation.findNavController
import androidx.viewpager2.widget.ViewPager2
import com.example.overtone.data.ChordGroup
import com.example.overtone.data.DifficultyLevel
import com.example.overtone.data.SingleChord
import com.example.overtone.recyclerview.ViewPagerAdapter
import com.google.android.material.tabs.TabLayout
import com.google.android.material.tabs.TabLayoutMediator
/**notice the fragment import to have access to view components*/
import kotlinx.android.synthetic.main.fragment_group_details.*
import java.util.*

class ChordGroupDetailsFrag : Fragment() {
    private var sentChordGroup: ChordGroup? = null
    /** testing image list remove when ready */
    private var chordDiagramIds: MutableList<Int> = mutableListOf()
    private var adapter:ViewPagerAdapter? = null
    private var namesIds = mutableMapOf<String,Int>()

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_group_details, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        initializeData()
        setUpViewPagerAdapter(chordDiagramIds)
        setUpTabLayout()
    }

    private fun retrievePassedFragmentData() {
        if (arguments != null) {
            val args = ChordGroupDetailsFragArgs.fromBundle(requireArguments())
            //receiving chord group data object that was clicked on
            sentChordGroup = args.chordGroupClickedOn
        }
    }

    private fun setChordDiagramIds() {
        chordDiagramIds = ArrayList()
        for(id in namesIds.values){
            chordDiagramIds.add(id)
        }
    }

    private fun setUpViewPagerAdapter(chordDiagramIds:MutableList<Int>) {
        adapter = ViewPagerAdapter(chordDiagramIds)
        viewPager.adapter = adapter
    }

    private fun setUpTabLayout(){
        //keeping material principles, let view be scrollable if there are 5+ components
        if(chordDiagramIds.size<5){
            tabLayout.tabMode = TabLayout.MODE_FIXED
        }
        var titles = namesIds.keys.toList()
        TabLayoutMediator(tabLayout,viewPager){tab, position ->
//            tab.text = "Tab ${position+1}"
            tab.text ="${titles[position]}"
        }.attach()
    }
    private fun mapNamesToIds(){
        for (singleChord in sentChordGroup?.chordList!!){
            //short hand for namesIds.put(singleChord.chordName,singleChord.imageID)
            namesIds[singleChord.chordName] = singleChord.imageID
        }
    }
    private fun initializeData() {
        retrievePassedFragmentData()
        mapNamesToIds()
        setChordDiagramIds()
    }



}